### ISSUE TITLE
[Add ASAP/URGENT] Limiting Disk Cache / Memory Usage / Access Speeds

### ISSUE THREAD

**Issue by SBSeed** [monero-project/monero-gui#1862]
massive issue which is causing 75% of the issues people are having...
this is leading to memory leaks, disk pagefile/cache freezing and more.
(information here is relative and percentages are not completely accurate since there is no current way to monitor every detail, or even with any accuracy, of monero hardware usages)

sadly most (if not all) of the problems that people are having with the crashing/freezing or partial of same of their systems no matter the OS or Hardware used, this also is not repeat NOT from physical hardware breakdowns (such as disk sector degradation)...

this is basically a false-positive of looking at logs and why ibdm blockchain data is/can become corrupted...

this type of uncontrolled usage of cache/memory is causing computers/servers to disconnect from the internet mid or late download as well as corruption of blockchain through partial or full freezing of OS and system because monerod.exe/monero-GUI/monero-blockchain-import.exe etc. ALL have this same problem (why this has never been addressed is beyond me) within the coding of the programs/apps.

because the programs/apps do not limit cache/memory/page file usage and clear/purge old information  as the program runs it overloads the cache/memory/page files and causes hell with any OS... currently the only way to clear/purge this information is by completely restarting the computer, this is a huge problem that needs to be fixed NOW!

there has to be hardcoding for limiting usage by setting a maximum percentage that the programs/apps use... this has been a misdiagnosis for YEARS and apparently no one even considered this as a potential issue or cause of most of the symptoms that people have been dealing with or complaining about connect-ability/blockchain data/etc.
through looking through old issues/bugs researching commands for CMD/console commands and personal investigation i have been able to find this and seeing how the programs work (especially on Win7 OS-PC) uses hardware...

if you use any webbrowsers or similar internet software at the same time as downloading/processing the blockchain through monero programs this is magnified by 50%, and is exponential in usage beyond this...
compairing downloading of usage's between just running the programs/apps with the blockchain and using browsers/etc. the time to crash/freeze is halved, the next program/browser used causes crashing/freeze down to 1/4 and so on...
this is also the case with using blockchain.raw (much MUCH less of a problem) when importing to imdb for monero programs to use to sync etc., using blockchain.raw is probably 75% (probably much higher) less problematic and less likely to cause these issues than downloading blockchain directly.

it has taken a week or so of looking into this (probably only a 2-3 days total) to notice this, the only reason i did notice is because of how i am able to view/process information in a completely different way to most people but it still surprises me to see no one else caught onto this before....

bottom line: these has to be addressed now or this will create worse problem with connecting and using monero in the future (this already effects the blockchain information and therefore effects recieving/sending of monero)

i would suggest adding controls for controlling usages of application of hardware and software...
- cache usage min 10% to max 75% (i think this would be good for most PCs that are not dedicated to monero usage)
- memory usage proper purging of information from physical memory and page files/virtual memory
- limitation on disk access speeds similar to cache usage (possible to do this for network interfacing as well)

**Comment by xiphon** [comment]
You must be kidding, but in the case you are serious:

> cache usage min 10% to max 75% (i think this would be good for most PCs that are not dedicated to monero usage)
> memory usage proper purging of information from physical memory and page files/virtual memory

That is what any OS and hardware are doing. Read OS docs to find out whether it supports per-process cache limiting and how to configure it.

> limitation on disk access speeds similar to cache usage (possible to do this for network interfacing as well)

Yep, disk access throttling might be implemented. 
But it won't resolve any problems if the underlying hardware doesn't provide enough disk I/O bandwidth and Monero daemon hits the limit.

**Comment by SBSeed** [comment]
@xiphon: why would you think i was joking?

@pazos:
- its not the CPU usage that is the issue, this can be limited by command settings before you run monerod with or without the GUI...
- monero-GUI and monerod ALREADY have a limit set on how much CPU usage happens
- WHEN the hardware caching and speeds are maxed this creates a idle cascade effect that causes issues and plays havoc with other programs
- the caching and accessing speeds will and DO top out and cause freezing since they are not controlled this can cause multiple issues to accure over time including creating what would appear to be a memory leak (this can easily be recreated on any PC)
- the same can and does happen with physical memory, the OS does a dump to physical memory and causes false reporting on physical memory usage which then causes further problems with operations and halts on the OS and hardware which causes total system freeze and you have to physically force shutdown

i might try a bountysource thing, however as things currently stand i have no money to donate (i am completely dependent on the state/gov because i am disabled) this is part of why i am even bothering to post anything (i would rather NOT waste my time and efforts) in the first place...
i am hoping that the Developers/programmers take this information into account and fix or add to monero project so that it becomes more stable, and so that more people can use it (or becomes more widely usable to laymen)...

it is becoming more and more essential to have alternatives to paper money as it is consistently losing value because it is made from air and is not finite.

**Comment by xmrdsc** [comment]
I am certain #1920 will 'fix' some performance issues some are seeing. I am going to close this issue for now and if there is any doubt about the inner workings of blockchain syncing, I suggest opening a PR on the [main repository](https://github.com/monero-project/monero), not the GUI repository. We are strictly for GUI related issues.

I thank you for the investigations. Please be aware you can also choose to fix the complaints you have yourself by submitting a pull request.

+invalid
